### ISSUE TITLE
New argument names

### ISSUE THREAD

**Issue by KnutJaegersberg** [AlbertoAlmuinha/neuralprophet#12]
New version of neuralprophet has new argument names. I repalced them in the parsnip-neuralprophet and load the pkg environment variable manually in a script. Otherwise, I could not install neuralprophet via conda install helper, nor otherwise. train_speed argument was droppped apparently. 

New function argument names: 

    def __init__(
        self,
        growth="linear",
        changepoints=None,
        n_changepoints=10,
        changepoints_range=0.9,
        trend_reg=0,
        trend_reg_threshold=False,
        yearly_seasonality="auto",
        weekly_seasonality="auto",
        daily_seasonality="auto",
        seasonality_mode="additive",
        seasonality_reg=0,
        n_forecasts=1,
        n_lags=0,
        num_hidden_layers=0,
        d_hidden=None,
        ar_sparsity=None,
        learning_rate=None,
        epochs=None,
        batch_size=None,
        loss_func="Huber",
        optimizer="AdamW",
        newer_samples_weight=2,
        newer_samples_start=0.0,
        impute_missing=True,
        collect_metrics=True,
        normalize="auto",
        global_normalization=False,
        global_time_normalization=True,
        unknown_data_normalization=False,
    ):

neural_prophet <- function(mode = "regression", growth = NULL, changepoints = NULL, n_changepoints = NULL, changepoint_range = NULL,
                          yearly_seasonality = NULL, weekly_seasonality = NULL, daily_seasonality = NULL,
                          season = NULL, trend_reg = NULL, trend_reg_threshold = NULL, seasonality_mode = NULL,
                          num_hidden_layers = NULL, d_hidden = NULL, ar_reg = NULL, learning_rate = NULL,
                          epochs = NULL, batch_size = NULL, loss_func = NULL,  normalize_y = NULL,
                          impute_missing = NULL, n_forecasts = NULL, n_lags = NULL, freq = NULL) {

    args <- list(

        # Prophet
        growth                    = rlang::enquo(growth),
        changepoints         = rlang::enquo(changepoints),
        n_changepoints           = rlang::enquo(n_changepoints),
        changepoint_range         = rlang::enquo(changepoint_range),
        yearly_seasonality        = rlang::enquo(yearly_seasonality),
        weekly_seasonality        = rlang::enquo(weekly_seasonality),
        daily_seasonality         = rlang::enquo(daily_seasonality),
        season                    = rlang::enquo(season),
        trend_reg                 = rlang::enquo(trend_reg),
        trend_reg_threshold       = rlang::enquo(trend_reg_threshold),
        seasonality_mode          = rlang::enquo(seasonality_mode),
        num_hidden_layers         = rlang::enquo(num_hidden_layers),
        d_hidden                  = rlang::enquo(d_hidden),
        ar_reg               = rlang::enquo(ar_reg),
        learning_rate                = rlang::enquo(learning_rate),
        epochs                    = rlang::enquo(epochs),
        batch_size                = rlang::enquo(batch_size),
        loss_func                 = rlang::enquo(loss_func),
        normalize_y               = rlang::enquo(normalize_y),
        impute_missing            = rlang::enquo(impute_missing),
        n_forecasts               = rlang::enquo(n_forecasts),
        n_lags                    = rlang::enquo(n_lags),
        freq                      = rlang::enquo(freq)
    )

worked for me.

**Comment by vanderWin** [comment]
Does this still work for you?

If so, could you describe what edits you made and where for the "def __init__(" replacement code you mentioned - would really like to get this package working.

**Comment by KnutJaegersberg** [comment]
I currently use Python for this, due to cuda updates on my rolling release system, haven't checked this R package anymore. 
You can compare the def init with the original one if you want to make it work (again?). 
It should come from the forecaster.py 

https://github.com/ourownstory/neural_prophet/blob/305a67e03d43138bb32bef59918b24a57473425a/neuralprophet/forecaster.py
